pkg/providers/provider.go

```diff
@@ -41,3 +43,21 @@ type Mapper interface {
 
 // VMStatus represents VM status
 type VMStatus string
+
+// SourceClientFactory provides default client factory implementation
+type SourceClientFactory struct{}
+
+// NewSourceClientFactory creates new factory
+func NewSourceClientFactory() *SourceClientFactory {
+	return &SourceClientFactory{}
+}
+
+// NewOvirtClient creates new Ovirt clients
+func (f *SourceClientFactory) NewOvirtClient(dataMap map[string]string) (aclient.VMClient, error) {
```

Comment:
I think it should not be located here - that package should be general and shouldn't know anything about its implementations.

Reply:
Do you suggest to move it to `OvirtProvider`?

Reply:
Yes, or to the potential provider factory.